internal/commands/server/list.go

```diff
@@ -46,14 +81,136 @@ func (s *listCommand) ExecuteWithoutArguments(exec commands.Executor) (output.Ou
 		})
 	}
 
+	columns := []output.TableColumn{
+		{Key: "uuid", Header: "UUID", Colour: ui.DefaultUUUIDColours},
+		{Key: "hostname", Header: "Hostname"},
+		{Key: "plan", Header: "Plan"},
+		{Key: "zone", Header: "Zone"},
+		{Key: "state", Header: "State"},
+	}
+
+	if ls.showIPAddresses != "none" {
+		ipaddressMap, err := getIPAddressesByServerUUID(servers, ls.showIPAddresses, svc)
+		if err != nil {
+			return nil, err
+		}
+
+		for i, row := range rows {
+			uuid := row[0].(string)
+
+			var listIpaddresses []listIPAddress
+			if serverIpaddresses, ok := ipaddressMap[uuid]; ok {
+				listIpaddresses = append(listIpaddresses, serverIpaddresses.IPAddresses...)
+			}
+			row = append(row[:3], row[2:]...)
+			row[2] = listIpaddresses
+			rows[i] = row
+		}
+		columns = append(columns[:3], columns[2:]...)
+		columns[2] = output.TableColumn{
+			Key:    "ip_addresses",
+			Header: "IP addresses",
+			Format: formatListIPAddresses,
+		}
+	}
+
 	return output.Table{
-		Columns: []output.TableColumn{
-			{Key: "uuid", Header: "UUID", Colour: ui.DefaultUUUIDColours},
-			{Key: "hostname", Header: "Hostname"},
-			{Key: "plan", Header: "Plan"},
-			{Key: "zone", Header: "Zone"},
-			{Key: "state", Header: "State"},
-		},
-		Rows: rows,
+		Columns: columns,
+		Rows:    rows,
 	}, nil
 }
+
+// getIPAddressesByServerUUID returns IP addresses grouped by server UUID. This function will be removed when server end-point response includes IP addresses.
+func getIPAddressesByServerUUID(servers *upcloud.Servers, accessType string, svc service.AllServices) (map[string]listServerIpaddresses, error) {
+	returnChan := make(chan listServerIpaddresses)
+	var wg sync.WaitGroup
+
+	for _, server := range servers.Servers {
+		wg.Add(1)
+		go func(server upcloud.Server) {
```

Comment:
Is this OK, or should the number of concurrent goroutines be limited?

Reply:
I don't think that is necessary to add that complexity in this context. Number of servers are usually far less that what can be considered high amount of goroutines